### PR TITLE
Get a listing of Webinars and Stationary events

### PR DESCRIPTION
**Pull request by mako321**

### CODE REVIEW

**Review by qunabu — APPROVED** [review]

File: app/Http/Controllers/EventAPIController.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public function index(ListEventRequest $request): JsonResponse
+    {
+        $orderDto = OrderDto::instantiateFromRequest($request);
+        $page = LengthAwarePaginator::resolveCurrentPage();
+        $perPage = $request->get('per_page') ?? ConstantEnum::PER_PAGE;
+
+        $events = $this->eventService->getEventsList($orderDto);
+        $result = new LengthAwarePaginator($events->forPage($page, $perPage), $events->count(), $perPage, $page, [
```
Comment (pa-cholek) [inline app/Http/Controllers/EventAPIController.php:31]: This retrieves all webinars and events and then extracts part of the returned collection and builds a paginator from it?

I think what we need to do instead, is to create a database view that represents searchable data in unified format, query that view, and return stuff from it using polymorphic relation.

There is a library for migrating views https://github.com/staudenmeir/laravel-migration-views

And we need a view that joins data from both tables using UNION
so something like
```
SELECT
webinars.id as event_id,
"EscolaLms\Webinars\Models\Webinar" as event_type,
webinars.active_from as start_date,
webinars.active_to as end_date,
UNION
SELECT 
stationary.id as event_id,
"EscolaLms\StationaryEvents\Models\StationaryEvent" as event_class,
stationary.start_date as start_date,
stationary.end_date as end_date,
```
etc, then query and paginate this view using Model (I'm pretty sure that we can just create a model that has $table='view_name' and it should work as if it was using a table) and that Model should have method
```
event(): MorphTo {
    return $this->morphTo()
} 
```
and then we need to just return JsonResource that extracts that morphed event data

I'm not sure if we ever did something like this but it's worth a try XD

**Review by pa-cholek — COMMENTED** [review]
This is not a scalable/efficient solution.

Alternatively to creating database view, we can just create a new table (`searchable_events`?) and Model; and insert data to this new table every time a Webinar or StationaryEvent is created (using listeners on webinar/stationary created event).
This table should have only id, event_id, event_type, and Model should have event():morphTo and then we can query it using whereHas('event', fn($query) => applycriteriatothisquery).

**Review by HerbertIV — APPROVED** [review]

**Review by qunabu — CHANGES_REQUESTED** [review]
please change as @pa-cholek suggests

File: app/Http/Resources/EventResource.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+use Illuminate\Http\Resources\Json\JsonResource;
+
+class EventResource extends JsonResource
+{
+    public function toArray($request)
+    {
+        $event = $this->event;
+
+        return $event instanceof StationaryEvent
+            ? StationaryEventResource::make($event)
```
Comment (pa-cholek) [inline app/Http/Resources/EventResource.php:17]: Shouldn't this be `StationaryEventResource::make($event)->toArray($request)`?

**Review by pa-cholek — APPROVED** [review]

**Pull request merged**